### DISCUSSION TITLE
Is an Accurate last position possible with Minimum Location Displacement?

### DISCUSSION THREAD

**Discussion by tre4b** [owntracks/android#1231] [Ideas & New Features]
Android App Version 2.4.7 (24704)
Android 11
Samsung Galaxy Note 10 lite
Google Play

Not an issue as such a question about configuration.   I am using significant move mode which if I am correct will do an update every 900seconds but ONLY if the device has moved more than 500 meters.

Having gone to work current position is showing a point that is about 400 meters away from my current location.   I wondered if there was a way for OwnTracks to send the accurate position if the same location comes up X number of times in the same place or within a smaller minimal distance regardless of whether it is less than the Minimal Location displacement?

I know that the issue is that minimal mode uses the cell towers which is not very accurate.  OwnTracks could grab from GPS for that final location though at the time,  I just do not know if it can be configured this way.

**Comment by growse** [comment]
I think the challenge here is coming up with a useful enough algorithm that works well for everyone, and isn't overly complex.

In theory, you could have some sort of heuristic that starts to measure how "stationary" the location is and periodically request a high accuracy location, but this might get a little complex to implement. Lots of questions like "What's the threshold for 'stationary'?", "How long does it have to be stationary for", which will all have different answers depending on the context.

Something that might be simpler to implement is to provide an Intent that lets OT fetch a single location at high-accuracy (GPS). That could then be called intermittently from an automation app. Alternatively, OT could itself have the option of periodically doing this, as long as it was less often than every 15 mins (because that's all you can do easily with the WorkManager). 

I can try and tinker around with some of these ideas behind an experimental feature flag if that would be useful?
